### DISCUSSION TITLE
Subscribe the bot to new posts from the channel

### DISCUSSION THREAD

**Discussion by tetelevm** [python-telegram-bot/python-telegram-bot#3231] [Q&A] [answered]
My bot needs to get information about posts in the channel. As far as I know, the bot can only get information about new posts, and I'm generally okay with a solution for new posts only. But I have a new problem - I can't receive messages from the channel.

What I did: I signed the bot to the channel (gave it admin rights) and inserted `print(update)` in the `app.process_update` method (inside the ptb). But nothing comes to me when a new post in the channel was posted.

Is there any example/manual of how to properly solve this problem? (And is it even possible?)

**Comment by harshil21** [comment]
Yes, it is possible to solve this easily. There is no need to use app.process_update for this. Just register a MessageHandler with the filter `filters.ALL` or `filters.TEXT`. And it will catch channel posts.

There is an example for this, [echobot.py](https://docs.python-telegram-bot.org/en/latest/examples.echobot.html#echobot-py)

**Comment by tetelevm** [comment, marked as answer]
Yay, it worked. It turned out that the problem was not with the handlers, but with using `allowed_updates` when starting the bot:
```python
await app.updater.start_polling(allowed_updates=[Update.CHANNEL_POST, Update.POLL_ANSWER])
```
But thanks anyway!
